stumpy/stumped.py

```diff
@@ -248,20 +253,41 @@ def stumped(dask_client, T_A, m, T_B=None, ignore_trivial=True, normalize=True,
                 T_B_subseq_isconstant_future,
                 diags_futures[i],
                 ignore_trivial,
+                k,
             )
         )
 
     results = dask_client.gather(futures)
-    profile, indices = results[0]
+    profile, indices, profile_L, indices_L, profile_R, indices_R = results[0]
+
     for i in range(1, len(hosts)):
-        P, I = results[i]
-        for col in range(P.shape[1]):  # pragma: no cover
-            cond = P[:, col] < profile[:, col]
-            profile[:, col] = np.where(cond, P[:, col], profile[:, col])
-            indices[:, col] = np.where(cond, I[:, col], indices[:, col])
-
-    out[:, 0] = profile[:, 0]
-    out[:, 1:4] = indices
+        P, I, PL, IL, PR, IR = results[i]
+        for j in range(l):
+            # Uodate profie[j]
+            for D, ind in zip(P[j], I[j]):
+                if D >= profile[j, -1]:
+                    break  # no need to update profile[j] from this point.
+                idx = np.searchsorted(profile[j], D, side="right")  # might be optimized
+                # with help of checkpoint idx from previous iteration.
+                profile[j, idx + 1 :] = profile[j, idx : k - 1]
```

Comment:
Why do you need `k - 1`? Why not just `-1`?

Reply:
> Why do you need `k - 1`? Why not just `-1`?

My bad! I should have used `-1`. Thanks for catching it!